File: countnotices.rb
require 'fileutils'

ARGV.each { |a|
	notices = Dir.entries("notices/#{a}")	# get all file names in notices/ and put them in an array
	tendays = 0
	onedays = 0
	reservations = 0
	invoices = 0
	notices.each { |file|
		File.chmod(0200, file)
		contents = File.read(file)
		if contents =~ /10 day overdue notice/i then
			tendays += 1
		end
		if contents =~ /first overdue notice/i then
			onedays += 1
		end
		if contents =~ /reservation notice/i then
			reservations += 1
		end
		if contents =~ /long overdue notice/i then
			invoices += 1
		end
	}
	puts "1-day overdues: #{onedays}"
	puts "10-day overdues: #{tendays}"
	puts "reservations: #{reservations}"
	puts "invoices: #{invoices}"
}

File: countnotices.py
# coding: interpy

import sys, os, re, time, yaml, pymssql

credentialspath = os.path.abspath(os.path.dirname(__file__)) + '/credentials.yml'
credentials = open(credentialspath, "r")
destinfo = yaml.load(credentials)
dbserver = destinfo['dbserver']
dbuser = destinfo['dbuser']
dbpassword = destinfo['dbpassword']
if len(sys.argv) == 1:
    today = time.strftime("%Y") + "-" + time.strftime("%m") + "-" + time.strftime("%d")
    sys.argv.append(today)
dates = sys.argv[1:]
for date in dates:
    rescount = 0
    overdueonecount = 0
    overduetencount = 0
    invoicecount = 0
    print date
    noticesdir = "notices/" + date + "/"
    notices = os.listdir(noticesdir)
    for filename in notices:
        with open(noticesdir + filename, 'r') as textfile:
            contents = textfile.read()
            if re.search('Reservation Notice:', contents):
                rescount += 1
            elif re.search('First Overdue Notice:', contents):
                overdueonecount += 1
            elif re.search('10 Day Overdue Notice:', contents):
                overduetencount += 1
            elif re.search('Long Overdue Invoice:', contents):
                invoicecount += 1
    totals = "Totals for #{date}:\nReservations: #{rescount}\nFirst Overdues: #{overdueonecount}\n10-day Overdues: #{overduetencount}\nInvoices: #{invoicecount}\n------"
    print totals
    try:
        conn = pymssql.connect(dbserver, dbuser, dbpassword, "vSmartNotices")
        cursor = conn.cursor()
        cursor.execute("insert into dbo.noticeRecords (noticedate, reservations, firstoverdues, tendayoverdues, invoices) values ('#{date}', #{rescount}, #{overdueonecount}, #{overduetencount}, #{invoicecount});")
        conn.commit()
        conn.close()
    except:
        print "Could not write to DB: ", sys.exc_info()
    else:
        print "Row written to DB."